### PR TITLE
Refactor start of org permissions setup flow

### PR DESCRIPTION
**Pull request by mnacos**
## Context

Just a small refactor based on conversations around https://github.com/DFE-Digital/apply-for-teacher-training/pull/2337

## Changes proposed in this pull request

Extract query for pending org permissions into a service.

## Guidance to review

Should be straightforward.

## Link to Trello card

https://trello.com/c/TfyXM7kv

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/controllers/provider_interface/provider_interface_controller.rb [hunk 1]
```diff
@@ -73,22 +73,8 @@ def check_provider_relationship_permissions
...
-        setup_at: nil,
-        training_provider: current_provider_user.providers,
-      )
-
-      if permissions.present?
-        auth = ProviderAuthorisation.new(actor: current_provider_user)
-
-        training_provider = permissions.training_provider
-        training_provider if auth.can_manage_organisation?(provider: training_provider)
+      if ProviderSetup.new(provider_user: current_provider_user).next_relationship_pending
```
Comment (duncanjbrown) [inline app/controllers/provider_interface/provider_interface_controller.rb:76]: Yes!

File: spec/services/provider_setup_spec.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      3.times.each do |idx|
+        expected_relationship = relationships[idx]
+        expect(next_relationship_pending).to eq(expected_relationship)
+        expected_relationship.update(setup_at: Time.current)
+      end
+
+      expect(next_relationship_pending).to be_nil
+    end
+
+    it 'returns nil if all manageable relationships have been set up' do
```
Comment (duncanjbrown) [inline spec/services/provider_setup_spec.rb:43]: I think this spec effectively duplicates the above — do we need it?

File: app/controllers/provider_interface/provider_interface_controller.rb (hunk 1 above)
Comment (duncanjbrown) [reply app/controllers/provider_interface/provider_interface_controller.rb:76]: We should have a follow-up to move the DSA code into here too

File: spec/services/provider_setup_spec.rb (hunk 2 above)
Comment (mnacos) [reply spec/services/provider_setup_spec.rb]: I guess it's covered by the previous one. Removing

**Pull request merged**